src/hierarchies/priors/nxig_prior_model.cc

```diff
@@ -0,0 +1,90 @@
+#include "nxig_prior_model.h"
+
+void NxIGPriorModel::initialize_hypers() {
+  if (prior->has_fixed_values()) {
+    // Set values
+    hypers->mean = prior->fixed_values().mean();
+    hypers->var = prior->fixed_values().var();
+    hypers->shape = prior->fixed_values().shape();
+    hypers->scale = prior->fixed_values().scale();
+    // Check validity
+    if (hypers->var <= 0) {
+      throw std::invalid_argument("Variance parameter must be > 0");
+    }
+    if (hypers->shape <= 0) {
+      throw std::invalid_argument("Shape parameter must be > 0");
+    }
+    if (hypers->scale <= 0) {
+      throw std::invalid_argument("scale parameter must be > 0");
+    }
+  } else {
+    throw std::invalid_argument("Unrecognized hierarchy prior");
+  }
+}
+
+double NxIGPriorModel::lpdf(const google::protobuf::Message &state_) {
+  auto &state = downcast_state(state_).uni_ls_state();
+  double target =
+      stan::math::normal_lpdf(state.mean(), hypers->mean, sqrt(hypers->var)) +
+      stan::math::inv_gamma_lpdf(state.var(), hypers->shape, hypers->scale);
+  return target;
+}
+
+std::shared_ptr<google::protobuf::Message> NxIGPriorModel::sample(
+    bayesmix::AlgorithmState::HierarchyHypers hier_hypers) {
+  auto &rng = bayesmix::Rng::Instance().get();
+  auto params = hier_hypers.nnxig_state();
+
+  double var = stan::math::inv_gamma_rng(params.shape(), params.scale(), rng);
+  double mean = stan::math::normal_rng(params.mean(), sqrt(params.var()), rng);
+
+  bayesmix::AlgorithmState::ClusterState state;
+  state.mutable_uni_ls_state()->set_mean(mean);
+  state.mutable_uni_ls_state()->set_var(var);
+  return std::make_shared<bayesmix::AlgorithmState::ClusterState>(state);
```

Comment:
I think that a cleaner solution would be

```
State::UniLS out;
out.mean = stan::math::normal_rng(params.mean(), sqrt(params.var()), rng);
out.var = stan::math::inv_gamma_rng(params.shape(), params.scale(), rng);
return out.to_proto()
```

Reply:
This is a nice improvement. We have the ``get_as_proto`` method in each ``State`` class, but we do not give any API to the state. Do you think we need to move towards a more structured object (for end users that want to define a new state)?

Reply:
I don't think that is essential right now, I would first merge this PR and then do the polishing